packages/x-charts/src/hooks/useReducedMotion.ts

```diff
@@ -29,10 +29,11 @@ export const useReducedMotion = () => {
 
     handleMediaChange(mql);
 
-    mql.addEventListener('change', handleMediaChange);
+    // MatchMedia is not fully supported in all environments, so we need to check if it exists before calling addEventListener
+    mql.addEventListener?.('change', handleMediaChange);
```

Comment:
Something doesn't make sense to me. If it's not fully supported in all environments, how is it possible for `useMediaQuery()` to rely on:

```suggestion
    mql.addEventListener('change', handleMediaChange);
```

I conclude that one of the two is wrong (either in Core or in X)

Actually `useColorScheme()` uses `mql.addListener`, why not `addEventListener`.

Reply:
Both [useMediaQuery.ts](https://github.com/mui/material-ui/blob/4c46022dc41281d20e6af26a32642ffef10839ee/packages/mui-system/src/useMediaQuery/useMediaQuery.ts) and [useMediaQuery.test.js](https://github.com/mui/material-ui/blob/3e9b63164b5df370393777a1263258e409490227/packages/mui-material/src/useMediaQuery/useMediaQuery.test.js) have ways of preventing the code from executing. 

In the main code we seem to be testing for the presence of `window.matchMedia`, which we do in our code as well, but our tests add a stub for it in [stubMatchMedia](https://github.com/mui/mui-x/blob/a31fb9f8c48f28b3076bb9e1c936fc2920530301/test/utils/pickers/misc.ts#L6)

On the `useMediaQuery.test.js` test themselves, it seems we use `css-mediaquery` to completely override the `matchMedia` behaviour implementing our own event emitter, which in turn never gets to the issue

Regarding using `.addListener` vs `.addEventListener`. `.addListener` is deprecated. Typescript marks it as deprecated and so does [MDN](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList)